src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -450,6 +466,28 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		return cell;
 	}
 
+	// Adds Paramters cell based on Notebook URI parameters
+	private addParametersCell(notebookUriParams: string, hasParameterCell: boolean, parameterCellIndex: number): void {
+		let uriParamsIndex = parameterCellIndex;
+		// Set new parameters as Injected Parameters cell after original parameter cell
+		if (hasParameterCell) {
+			uriParamsIndex = parameterCellIndex + 1;
+			this.addUriParameterCell(uriParamsIndex, notebookUriParams);
+			this.cells[uriParamsIndex].isInjectedParameter = true;
+			this.cells[uriParamsIndex].source = ['# Injected-Parameters\n'].concat(this.cells[uriParamsIndex].source);
+		} else {
+			// Set new parameters as the parameters cell as the first cell in the notebook
+			this.addUriParameterCell(uriParamsIndex, notebookUriParams);
+			this.cells[uriParamsIndex].isParameter = true;
+		}
+	}
+
+	// Add new notebook URI parameters cell to notebook
```

Comment:
Use actual js doc (///) comments for all function comments please

Reply:
It is and should be - you don't get intellisense with just normal // comments so that kind of defeats the purpose of having function doc comments in the first place. 

(and yeah it's /**, not ///. Darn C# conventions messing me up again)